src/lnp/session/noise/conduit.rs

```diff
@@ -0,0 +1,630 @@
+use super::{chacha, hkdf};
+use crate::lnp::session::transcoders::{Decrypt, Encrypt, Transcode};
+use amplify::Bipolar;
+
+use super::handshake::HandshakeError;
+use chacha20poly1305::aead::Error;
+use std::borrow::Borrow;
+
+pub type SymmetricKey = [u8; 32];
+
+/// Maximum Lightning message data length according to
+/// [BOLT-8](https://github.com/lightningnetwork/lightning-rfc/blob/v1.0/08-transport.md#lightning-message-specification)
+/// and [BOLT-1](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#lightning-message-format):
+const LN_MAX_MSG_LEN: usize = ::std::u16::MAX as usize; // Must be equal to 65535
+
+const MESSAGE_LENGTH_HEADER_SIZE: usize = 2;
+const TAGGED_MESSAGE_LENGTH_HEADER_SIZE: usize =
+    MESSAGE_LENGTH_HEADER_SIZE + chacha::TAG_SIZE;
+
+const KEY_ROTATION_INDEX: u32 = 1000;
+
+pub struct Encryptor {
+    sending_key: SymmetricKey,
+    sending_chaining_key: SymmetricKey,
+    sending_nonce: u32,
+}
+
+impl Encryptor {
+    pub fn encrypt(&mut self, buffer: &[u8]) -> Result<Vec<u8>, Error> {
+        if buffer.len() > LN_MAX_MSG_LEN {
+            panic!("Attempted to encrypt message longer than 65535 bytes!");
```

Comment:
No panic should happen here but error reporting. I know that error would not propagate, but we don't need it to propagate: we have message size check at the transport level which happens before the encryption: https://github.com/LNP-BP/rust-lnpbp/blob/master/src/lnp/transport/ftcp.rs#L158

Reply:
Oh, it seems this was a bug. the `ftcp` msg size check is as below
```
if len > super::MAX_FRAME_SIZE {
            return Err(Error::OversizedFrame(len));
        }
```
where `MAX_FRAME_SIZE = 0xFFFF + 34`. But in `conduit` above the encryption length is being checked for `LN_MAX_MSG_LEN` which is `0xFFFF`. So even if the size was correct as per `ftcp` check, it would have panicked here.

My suggestion is we get rid of the check here and add documentation warning users that they should check for max msg length before passing for encryption here.

Reply:
Also, should we use `MAX_FRAME_SIZE` which is `0xFFFF + 34` in all places which have check against `LN_MAX_MSG_LEN`? It should depend on the context but my question is, in general, we should check for the maximum allowed frame size which is `0xFFFF + 34` instead of the maximum allowed msg size which is `0xFFFF`, right?

Reply:
> My suggestion is we get rid of the check here and add documentation warning users that they should check for max msg length before passing for encryption here.

No need, I think. While handshake is part of BOLT-8, it's in the other part and related to LN message encoding but not to a specific encryption scheme. I am pretty much sure that this noise encryption can be used in other non-LN cases which do not require message size limit